### PR TITLE
Add tests for shell (squashed)

### PR DESCRIPTION
**Pull request by holdenk**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,17 +10,25 @@ install:
...
+  - "pylint -E sparklingpandas --extension-pkg-whitelist=numpy"
+  - echo $PATH
+  - export LOCATION=`pip show sparklingpandas|grep Location| cut -d ' ' -f 2`
+  - echo $LOCATION
+  - ls $LOCATION/../
+  - ls $LOCATION/../../
+  - ls $LOCATION/../../../
+  - SHELL_LOCATION=$LOCATON/../../../
+  - cd /tmp; echo "exit(0)" | sparklingpandashell
+
```
Comment (jhlch) [inline .travis.yml]: Should end in a new line.

File: MANIFEST.in [hunk 2]
```diff
@@ -1,4 +1,5 @@
 include *.txt
 include *.md
+include sparklingpandashell
 include current-release/*.jar
-recursive-include docs *.md
+recursive-include docs *.md
```
Comment (jhlch) [inline MANIFEST.in]: new line.

File: .travis.yml [hunk 3]
```diff
@@ -10,17 +10,25 @@ install:
...
-  - "export SPARK_CONF_DIR=./sparklingpandas/test/resources/conf"
-  - "nosetests --logging-level=ERROR --detailed-errors --verbosity=2 --with-coverage --cover-html-dir=./htmlcov --cover-package=sparklingpandas"
+  - "nosetests --logging-level=INFO --detailed-errors --verbosity=2 --with-coverage --cover-html-dir=./htmlcov --cover-package=sparklingpandas"
   - "pep8 --ignore=E402 sparklingpandas/"
-  - "pylint -E sparklingpandas --extension-pkg-whitelist=numpy --disable=no-member"
+  - "pylint -E sparklingpandas --extension-pkg-whitelist=numpy"
+  - echo $PATH
+  - export LOCATION=`pip show sparklingpandas|grep Location| cut -d ' ' -f 2`
+  - echo $LOCATION
+  - ls $LOCATION/../
```
Comment (jhlch) [inline .travis.yml]: What is up with all the ls's? Why are they needed in travis? Were they just for debugging and if so can they be removed now?

File: sparklingpandas/__init__.py [hunk 4]
```diff
@@ -33,8 +33,14 @@
...
-    JAR_FILE = 'sparklingpandas_2.10-' + VERSION + '-SNAPSHOT.jar'
-    jar = os.path.join(sys.prefix, 'jar', JAR_FILE)
+    JAR_FILE = 'sparklingpandas-assembly-' + VERSION + '-SNAPSHOT.jar'
+    my_location = os.path.dirname(os.path.realpath(__file__))
+    prefixes = [os.path.join(my_location, '../target/scala-2.10/'),
+                os.path.join(my_location, '../curret-release/'),
+                os.path.join(sys.prefix, "local/jars/")]
+    jars = map(lambda prefix: os.path.join(prefix, JAR_FILE), prefixes)
+    print jars
+    jar = filter(lambda path: os.path.exists(path), jars)[0]
```
Comment (jhlch) [inline sparklingpandas/__init__.py]: filter is deprecated, you can use list comprehensions with a filtering clause instead.

File: sparklingpandas/__init__.py [hunk 5]
```diff
@@ -33,8 +33,14 @@
...
     VERSION = '0.0.4'
-    JAR_FILE = 'sparklingpandas_2.10-' + VERSION + '-SNAPSHOT.jar'
-    jar = os.path.join(sys.prefix, 'jar', JAR_FILE)
+    JAR_FILE = 'sparklingpandas-assembly-' + VERSION + '-SNAPSHOT.jar'
+    my_location = os.path.dirname(os.path.realpath(__file__))
+    prefixes = [os.path.join(my_location, '../target/scala-2.10/'),
+                os.path.join(my_location, '../curret-release/'),
+                os.path.join(sys.prefix, "local/jars/")]
+    jars = map(lambda prefix: os.path.join(prefix, JAR_FILE), prefixes)
+    print jars
```
Comment (jhlch) [inline sparklingpandas/__init__.py]: remove print statement

File: sparklingpandas/__init__.py [hunk 6]
```diff
@@ -33,8 +33,14 @@
 
 if 'IS_TEST' not in os.environ:
     VERSION = '0.0.4'
-    JAR_FILE = 'sparklingpandas_2.10-' + VERSION + '-SNAPSHOT.jar'
-    jar = os.path.join(sys.prefix, 'jar', JAR_FILE)
+    JAR_FILE = 'sparklingpandas-assembly-' + VERSION + '-SNAPSHOT.jar'
+    my_location = os.path.dirname(os.path.realpath(__file__))
+    prefixes = [os.path.join(my_location, '../target/scala-2.10/'),
```
Comment (jhlch) [inline sparklingpandas/__init__.py]: Add a comment to explain what you are doing with all the possibilities.

File: sparklingpandas/__init__.py [hunk 7]
```diff
@@ -33,8 +33,14 @@
...
+                os.path.join(my_location, '../curret-release/'),
```
Comment (jhlch) [inline sparklingpandas/__init__.py]: spelling

File: sparklingpandas/pcontext.py [hunk 8]
```diff
@@ -39,8 +39,10 @@ def __init__(self, spark_context, sql_ctx=None):
         """
         self.spark_ctx = spark_context
         if sql_ctx:
+            print "Using provided sql context"
```
Comment (jhlch) [inline sparklingpandas/pcontext.py]: We should using logging instead of print statements.

File: sparklingpandashell [hunk 9]
```diff
@@ -6,11 +6,14 @@ fi
...
-fi
 set -x
 SPARK_EXECUTOR_MEMORY=${SPARK_EXECUTOR_MEMORY-"2000m"}
-JARS=`ls ./target/scala-2.10/*.jar`
 SPARK_JAVA_OPTS=${SPARK_JAVA_OPTS-"-Dspark.executor.memory=$SPARK_EXECUTOR_MEMORY -cp $CLASSPATH"}
-PYTHONSTARTUP=./sparklingpandas/shell.py PYTHON_PATH=`pwd`:$PTHYON_PATH $SPARK_HOME/bin/pyspark --jars $JARS --driver-class-path $JARS $@
+PIP_LOCATION=`pip show sparklingpandas|grep Location| cut -d ' ' -f 2`
+# If we have the shell relative to us use it, otherwise use the pip determined one
+if [ -e "./sparklingpandas/shell.py" ]; then
+export PYTHONSTARTUP="./sparklingpandas/shell.py"
```
Comment (jhlch) [inline sparklingpandashell]: indentation

**Comment by holdenk** [comment]
Merged in.

**Pull request closed without merging**